pkg/pipeline/PipelineBuilder.go

```diff
@@ -1722,6 +1591,204 @@ func (impl PipelineBuilderImpl) GetArtifactsByCDPipeline(cdPipelineId int, stage
+	return ciArtifactsResponse, nil
+}
+
+func (impl PipelineBuilderImpl) GetArtifactsForDeployCdStage(cdPipelineId int, parentId int, parentType bean2.CdWorkflowType) (bean.CiArtifactResponse, error) {
+	var ciArtifacts []bean.CiArtifactBean
+	var ciArtifactsResponse bean.CiArtifactResponse
+	pipeline, err := impl.pipelineRepository.FindById(cdPipelineId)
+	if err != nil && err != pg.ErrNoRows {
+		impl.logger.Errorw("Error in getting cd pipeline details", err, "cdPipelineId", cdPipelineId)
+	}
+	var artifacts []repository.CiArtifact
+	if len(pipeline.PreStageConfig) > 0 {
+		artifacts, err = impl.ciArtifactRepository.GetArtifactsByCDPipelineAndRunnerType(cdPipelineId, bean2.CD_WORKFLOW_TYPE_PRE)
+		if err != nil {
+			impl.logger.Errorw("error in getting artifacts for cd", "err", err, "parentStage", bean2.CD_WORKFLOW_TYPE_PRE, "childStage", bean2.CD_WORKFLOW_TYPE_DEPLOY)
+			return ciArtifactsResponse, err
+		}
+	} else if parentType == bean2.CD_WORKFLOW_TYPE_POST {
+		artifacts, err = impl.ciArtifactRepository.GetArtifactsByCDPipelineAndRunnerType(parentId, bean2.CD_WORKFLOW_TYPE_POST)
+		if err != nil {
+			impl.logger.Errorw("error in getting artifacts for cd", "err", err, "parentCdId", parentId, "parentStage", parentType, "childStage", bean2.CD_WORKFLOW_TYPE_DEPLOY)
+			return ciArtifactsResponse, err
+		}
+	} else if parentType == bean2.CD_WORKFLOW_TYPE_DEPLOY {
+		artifacts, err = impl.ciArtifactRepository.GetArtifactsByCDPipelineAndRunnerType(parentId, bean2.CD_WORKFLOW_TYPE_DEPLOY)
+		if err != nil {
+			impl.logger.Errorw("error in getting artifacts for cd", "err", err, "parentCdId", parentId, "parentStage", parentType, "childStage", bean2.CD_WORKFLOW_TYPE_DEPLOY)
+			return ciArtifactsResponse, err
+		}
+	} else {
+		artifacts, err = impl.ciArtifactRepository.GetArtifactsByCDPipeline(cdPipelineId)
+		if err != nil {
+			impl.logger.Errorw("error in getting artifacts for cd", "err", err, "parentStage", appWorkflow.CIPIPELINE, "childStage", bean2.CD_WORKFLOW_TYPE_DEPLOY)
+			return ciArtifactsResponse, err
+		}
+	}
+	latestFound := false
+	artifactMap := make(map[int]int)
+	for _, artifact := range artifacts {
+		artifactMap[artifact.Id] = artifact.Id
+		mInfo, err := parseMaterialInfo([]byte(artifact.MaterialInfo), artifact.DataSource)
+		if err != nil {
+			mInfo = []byte("[]")
+			impl.logger.Errorw("Error in parsing artifact material info", "err", err, "artifact", artifact)
+		}
+
+		ciArtifacts = append(ciArtifacts, bean.CiArtifactBean{
+			Id:           artifact.Id,
+			Image:        artifact.Image,
+			ImageDigest:  artifact.ImageDigest,
+			MaterialInfo: mInfo,
+			DeployedTime: formatDate(artifact.DeployedTime, bean.LayoutRFC3339),
+			Deployed:     artifact.Deployed,
+			Latest:       artifact.Latest,
+		})
+		if artifact.Latest == true {
+			latestFound = true
+		}
+	}
+
+	//start adding deployed items
+	deployedItemsCiArtifactResponse, err := impl.GetArtifactsForPostCdStage(cdPipelineId, latestFound)
```

Comment:
why are we not calling it in case of preCdStage? there also we need all deployed images